core/dbt/cli/flags.py

```diff
@@ -49,13 +49,33 @@ def convert_config(config_name, config_value):
     # This function should take care of converting the values from config and original
     # set_from_args to the correct type
     ret = config_value
-    if config_name.lower() == "warn_error_options":
+    if config_name.lower() == "warn_error_options" and type(config_value) == dict:
         ret = WarnErrorOptions(
             include=config_value.get("include", []), exclude=config_value.get("exclude", [])
         )
     return ret
 
 
+def args_to_context(args):
```

Comment:
What is the intended usage of this method? Is it just to create a click context that can be used to construct a `Flags` instance, or would we also want callers invoking the click command on the built context? If it's just the first use case, perhaps we could add this as a class method on Flags instead, and get eventually use it to get rid of `set_from_args`?

Reply:
Yes it is, and that's what I mean by Flag object is leaky.
The only place that uses this right now is in `dbt-rpc`, and we are gonna deprecate it in 1.6. So I am thinking of merging this change for now to make the 1.5 beta release, and do a revisit of the whole flags object before final cut of 1.5